src/orange/detail/UniverseIndexer.hh

```diff
@@ -83,7 +83,7 @@ class UniverseIndexer
         = Collection<size_type, Ownership::const_reference, MemSpace::native>;
     using AllVals = AllItems<size_type, MemSpace::native>;
     using SizeId = OpaqueId<size_type>;
-    using SpanIter = Span<size_type const>::const_iterator;
+    using SpanIter = typename DataRef::SpanConstT::const_iterator;
```

Comment:
This is sweet: it means with effectively no change to our code, the `upper_bound` implementation we borrowed from libcxx now automatically uses `__ldg` to converge on the data: @tmdelellis reported that a hand-rolled lower-bound written with `ldg` improved their performance substantially.

Reply:
yeah, since `LdgIterator` complies with all the requirements it can be used in most places where a standard iterator is expected, it should work with thrust as well.